http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/redirect/HTTPClientRedirectTestCase.java

```diff
@@ -0,0 +1,565 @@
+                        senderConfiguration);
+
+    }
+
+    /**
+     * Check whether, redirect request is written to the backend when a redirect response is received.
+     *
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    @Test
+    public void unitTestForRedirectHandler() throws URISyntaxException, IOException {
+        EmbeddedChannel embeddedChannel = new EmbeddedChannel();
+        embeddedChannel.pipeline().addLast(new HttpResponseDecoder());
+        embeddedChannel.pipeline().addLast(new HttpRequestEncoder());
+        embeddedChannel.pipeline().addLast(new RedirectHandler(null, false, 5));
+        HttpResponse response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.TEMPORARY_REDIRECT,
+                Unpooled.EMPTY_BUFFER);
+        response.headers().set(HttpHeaders.Names.LOCATION, FINAL_DESTINATION);
+        embeddedChannel.attr(Constants.ORIGINAL_REQUEST)
+                .set(createHttpRequest(Constants.HTTP_GET_METHOD, FINAL_DESTINATION));
+        embeddedChannel.writeInbound(response);
+        embeddedChannel.writeInbound(LastHttpContent.EMPTY_LAST_CONTENT);
+        assertNotNull(embeddedChannel.readOutbound());
+    }
+
+    /**
+     * When the maximum redirect count reached, channel should not do any more redirects.
+     *
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    @Test
+    public void unitTestForRedirectLoop() throws URISyntaxException, IOException {
+        EmbeddedChannel embeddedChannel = new EmbeddedChannel();
+        embeddedChannel.pipeline().addLast(new HttpResponseDecoder());
+        embeddedChannel.pipeline().addLast(new HttpRequestEncoder());
+        embeddedChannel.pipeline()
+                .addLast(Constants.IDLE_STATE_HANDLER, new IdleStateHandler(50000, 50000, 0, TimeUnit.MILLISECONDS));
+        embeddedChannel.pipeline().addLast(new RedirectHandler(null, false, 5, null, false));
+        HttpResponse response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.TEMPORARY_REDIRECT,
+                Unpooled.EMPTY_BUFFER);
+        response.headers().set(HttpHeaders.Names.LOCATION, FINAL_DESTINATION);
+        embeddedChannel.attr(Constants.ORIGINAL_REQUEST)
+                .set(createHttpRequest(Constants.HTTP_GET_METHOD, FINAL_DESTINATION));
+        embeddedChannel.attr(Constants.RESPONSE_FUTURE_OF_ORIGINAL_CHANNEL).set(new HttpResponseFutureImpl());
+        TargetChannel targetChannel = new TargetChannel(null, null);
+        targetChannel.setChannel(embeddedChannel);
+        embeddedChannel.attr(Constants.TARGET_CHANNEL_REFERENCE).set(targetChannel);
+        embeddedChannel.attr(Constants.REDIRECT_COUNT).set(5);
+        embeddedChannel.writeInbound(response);
+        embeddedChannel.writeInbound(LastHttpContent.EMPTY_LAST_CONTENT);
+        assertNull(embeddedChannel.readOutbound());
```

Comment:
Shall we add the assert messages?

Reply:
Done >> https://github.com/wso2/carbon-transports/pull/412/files#diff-44d72deb2696196bfa93f63a55a2069aL123